File: java/client/src/main/java/com/github/fernthedev/client/CLogger.java
package com.github.fernthedev.client;

import com.github.fernthedev.core.StaticHandler;
import lombok.NonNull;
import lombok.RequiredArgsConstructor;
import org.slf4j.Logger;


@RequiredArgsConstructor
public class CLogger implements ILogManager {
    @NonNull
    private Logger logger;

    @Override
    public void log(String log) {
        logger.info(log);
    }

    @Override
    public void logError(String log, Throwable e) {
        logger.error(log,e);
    }

    @Override
    public void info(String s) {
        log(s);
    }

    @Override
    public void debug(String s) {
        if(StaticHandler.isDebug) {
            log("[DEBUG] " + s);
        }
    }

    @Override
    public void error(String s) {
        logger.error(s);
    }
}

File: java/client/src/main/java/com/github/fernthedev/lightchat/client/CLogger.kt
package com.github.fernthedev.lightchat.client

import com.github.fernthedev.lightchat.core.StaticHandler.isDebug
import org.slf4j.Logger

class CLogger(private val logger: Logger) : ILogManager {
    override fun log(log: String?) {
        logger.info(log)
    }

    override fun logError(log: String?, e: Throwable?) {
        logger.error(log, e)
    }

    override fun info(s: String?) {
        log(s)
    }

    override fun debug(s: String) {
        if (isDebug()) {
            log("[DEBUG] $s")
        }
    }

    override fun error(s: String?) {
        logger.error(s)
    }
}